frontend/src/utils/validation.ts

```diff
@@ -33,9 +33,14 @@ export const assertValidHexString = (hex: string, length: number, errorMsg?: str
   }
 };
 
-export const assertNoConfusables = (name: string, errorMsg?: string) => {
-  const hasConfusables = isConfusing(name);
-  if (hasConfusables) {
-    throw new Error(errorMsg || 'Name contains confusables');
+export const assertValidEnsName = (name: string) => {
+  try {
+    ens_normalize(name);
+  } catch (err) {
+    throw new Error(
+      `We detected a${
+        (err as Error)?.message.split(':')[1]
```

Comment:
What's going on here, the way this works is that if it can't be normalized the special character is immediately after the colon in the error message? Would be good to show a sample error message as a comment for maintainability purposes, since relying on the format of a deps error message feels a bit flaky

Reply:
Yes, will do. The docs say the error type can be extracted after the first colon of the error message. From some testing this pattern seems to be stable. I will add some example comments.

Reply:
I see you added:
```typescript
// A couple example error messages from ens-normalize:
// ‌hi.eth - Invalid label "{200C}hi"‎: disallowed character: {200C}
// aα.eth - Invalid label "aα"‎: illegal mixture: Latin + Greek "α"‎ {3B1}
```

but:

```typescript
x = `Invalid label "{200C}hi"‎: disallowed character: {200C}`
x.split(':')[1] // <-- returns ' disallowed character'
```

Am I misinterpreting the comment?

Reply:
Ohh I see, we're just showing an error of `We detected a disallowed character in the ENS name.`, but we are not showing what the disallowed character is

Wdyt about changing the error message to `The given ENS name is invalid due to: '${x}'. Check the ENS name in order to avoid a potential scam.` Where `x` would be `disallowed character: {200C}` or `illegal mixture: Latin + Greek "α"‎ {3B1}` with the above examples. This way the error is more specific and helpful

Reply:
Yes, definitely I will add that

Reply:
https://github.com/ScopeLift/umbra-protocol/pull/555/commits/ae44c46020baeeecd16b68bf7e06276a24e6d01c